spec/controllers/automated_tests_controller_spec.rb

```diff
@@ -240,98 +234,93 @@
+      before { get_as user, :student_interface, params: { id: assignment.id } }
+      it('should respond with 403') { expect(response.status).to eq 403 }
     end
     context 'POST execute_test_run' do
-      before { post_as grader, :execute_test_run, params: { id: 1 } }
-      it('should respond with 404') { expect(response.status).to eq 404 }
+      before { post_as user, :execute_test_run, params: { id: assignment.id } }
+      it('should respond with 403') { expect(response.status).to eq 403 }
     end
     context 'GET get_test_runs_students' do
-      before { post_as grader, :get_test_runs_students, params: params }
-      it('should respond with 404') { expect(response.status).to eq 404 }
-    end
-    context 'GET populate_autotest_manager' do
-      before { get_as grader, :populate_autotest_manager, params: params }
-      it('should respond with 404') { expect(response.status).to eq 404 }
+      before { post_as user, :get_test_runs_students, params: params }
+      it('should respond with 403') { expect(response.status).to eq 403 }
     end
-    context 'GET download_file' do
-      before { get_as grader, :download_file, params: params }
-      it('should respond with 404') { expect(response.status).to eq 404 }
+  end
+  context 'as a student' do
+    let!(:user) { create :student }
+    context 'GET student_interface' do
+      before { get_as user, :student_interface, params: params }
+      # TODO: write tests
     end
-    context 'GET download_files' do
-      before { get_as grader, :download_files, params: params }
-      it('should respond with 404') { expect(response.status).to eq 404 }
+    context 'POST execute_test_run' do
+      before { post_as user, :execute_test_run, params: params }
+      # TODO: write tests
     end
-    context 'POST upload_files' do
-      before { post_as grader, :upload_files, params: params }
-      it('should respond with 404') { expect(response.status).to eq 404 }
+    context 'GET get_test_runs_students' do
+      before { post_as user, :get_test_runs_students, params: params }
+      # TODO: write tests
     end
-    context 'GET download_specs' do
-      before { get_as grader, :download_specs, params: params }
-      it('should respond with 404') { expect(response.status).to eq 404 }
+    context 'When student trying to manage automated testing' do
+      include_examples 'An unauthorized user managing automated testing'
     end
-    context 'POST upload_specs' do
-      before { post_as grader, :upload_specs, params: params }
-      it('should respond with 404') { expect(response.status).to eq 404 }
+  end
+  describe 'an authenticated admin' do
+    let!(:user) { create(:admin) }
+    include_examples 'An authorized admin and grader managing automated testing'
+  end
+
+  describe 'When the grader is allowed to download grades report and populate course summary' do
+    let!(:user) { create(:ta) }
+    before do
+      user.grader_permission.manage_assessments = true
```

Comment:
it would be nice to change the ta factory so that you could create tas with grader permissions already set (you could pass in the permissions you want when you call `create` instead of having to update the permissions in a `before` block)